packages/protocol/test/governance/downtime_slasher_intervals.ts

```diff
@@ -0,0 +1,731 @@
+        const startBlock = (epoch - 1) * epochBlockSize + 1
+        // All the other block are good
+        await presetParentSealForBlocks(startBlock + 1, slashableDowntime - 1, [bitmapVI0])
+        // first block with everything signed
+        await presetParentSealForBlocks(startBlock, 1, [bitmapVI99])
+        const slotArrays = await calculateEverySlot(startBlock)
+        await assertRevert(
+          slasher.slash(
+            slotArrays.startSlots,
+            slotArrays.endSlots,
+            [99],
+            0,
+            [],
+            [],
+            [],
+            [],
+            [],
+            []
+          ),
+          'not down'
+        )
+      })
+      describe('when the validator was down', () => {
+        let startBlock: number
+        beforeEach(async () => {
+          startBlock = (epoch - 1) * epochBlockSize + 1
+          await presetParentSealForBlocks(startBlock, slashableDowntime, [
+            bitmapWithoutValidator[validatorIndexInEpoch],
+          ])
+        })
+        describe('when the slots cover the SlashableDowntime window', () => {
+          it('success if slots overlap', async () => {
+            const startSlots = [startBlock, startBlock + 2]
+            const endSlots = [
+              startBlock + slashableDowntime - 3,
+              startBlock + slashableDowntime - 1,
+            ]
+            await generateProofs(startSlots, endSlots)
+            await slasher.slash(
+              startSlots,
+              endSlots,
+              [validatorIndexInEpoch],
+              0,
+              [],
+              [],
+              [],
+              [],
+              [],
+              []
+            )
+            const balance = await mockLockedGold.accountTotalLockedGold(validatorList[0])
+            assert.equal(balance.toNumber(), 40000)
+          })
+          it('success if slots cover more than the SlashableDowntime window', async () => {
+            const startSlots = [startBlock, startBlock + intervalSize]
+            const endSlots = [startBlock + intervalSize - 1, startBlock + slashableDowntime + 3]
+
+            for (let i = 0; i < startSlots.length; i += 1) {
+              await presetParentSealForBlocks(startSlots[i], endSlots[i] - startSlots[i] + 1, [
+                bitmapWithoutValidator[validatorIndexInEpoch],
+              ])
+            }
+            await generateProofs(startSlots, endSlots)
+            await slasher.slash(
+              startSlots,
+              endSlots,
+              [validatorIndexInEpoch],
+              0,
+              [],
+              [],
+              [],
+              [],
+              [],
+              []
+            )
+            const balance = await mockLockedGold.accountTotalLockedGold(validatorList[0])
+            assert.equal(balance.toNumber(), 40000)
+          })
+          it('fails if endSlot(i) is not between [startSlot(i+1)-1, endSlot(i+1)]', async () => {
```

Comment:
Would this be clearer as "fails if the slots are not ordered" or "not contiguous"?